gcp_variant_transforms/libs/processed_variant.py

```diff
@@ -164,10 +169,12 @@ class ProcessedVariantFactory(object):
   """
   def __init__(
       self,
-      header_fields,
-      split_alternate_allele_info_fields=True,
-      annotation_fields=None):
-    # type: (vcf_header_parser.HeaderFields, bool, List[str]) -> None
+      header_fields,  # type: vcf_header_parser.HeaderFields
+      split_alternate_allele_info_fields=True,  # type: bool
+      annotation_fields=None,  # type: List[str]
+      counter_factory=beam_util.BaseCounterFactory()
```

Comment:
Please do not use mutable objects as default values as it can have unexpected characteristics: http://docs.python-guide.org/en/latest/writing/gotchas/
You can set this to None and then initialize to `counter_factory or beam_util.BaseCounterFactory()` in the method. Or just make it a required field?

Reply:
Thinking a little more about this, while BaseCounterFactory is semantically immutable but I suppose in Python a rogue function can mutate anything passed to it (e.g., by accessing __dict__) and so cause confusing bugs (for that very function/class). If that is the concern, I can make counter_factory's default value None and check for None *only in constructor*. Still I will keep BaseCounter as a no-op implementation and if no counter_factory is passed to the class, use instances of BaseCounter to initialize member counters. Please let me know which way you prefer (so in that case BaseCounterFactory will become CounterFactoryInterface with no implementation but BaseCounter will be kept the way it is).

Reply:
For the record, I went with the second option that I suggested to make sure this is safe. The code is not as clean as it used to be but I think that is fine. If you have more comments, please let me know and I will fix in future PRs.